add optional params to loadResources for more flexibility

jab opened the pull request:
Fixes #75.

jab commented:
(removed the negotiateLanguage stuff from this PR for [reasons](https://github.com/PolymerElements/app-localize-behavior/pull/109#issuecomment-328684789), so this is now just all about the optional params to loadResources for more flexibility)

notwaldorf commented on app-localize-behavior.html:
```diff
@@ -185,19 +212,24 @@
     var request = proto.__localizationCache.requests[path];
+
+    function onRequestResponse(event) {
+      this.__onRequestResponse(event, language, merge, preprocessor);
```
Where are these parameters coming from? why would they not always be null? do you mean `this.language` etc?

notwaldorf replied:
Oh I see. I would just inline it in the 2 call sites, for clarity

notwaldorf commented on app-localize-behavior.html:
```diff
@@ -52,6 +52,33 @@
+* `merge = true`, the fetched resources will be merged into any existing
+* resources rather than clobbering them. Object.assign is used for the merge,
+* but you can also pass your own merge function to use that instead.
+* Finally, if you pass an optional fourth parameter, `preprocessor`, that
```
Uhhh it feels like we're throwing a lot of stuff in. Why is `preprocessor` needed? Why is `merge` needed? Do you actually have examples when you would need this, or "just in case" ?

jab replied:
The `preprocessor` param was an attempt to resolve #26 and #103 while I was at it. Since I currently only need #75 fixed, I'm happy to drop `preprocessor` from this PR and just did so in 2e60b42.

notwaldorf commented on app-localize-behavior.html:
```diff
@@ -52,6 +52,33 @@
+* If the resources stored in your external file are for a single language and
+* so are not nested inside any language keys, you can pass an optional
+* `language` parameter to store the fetched resources inside that key.
+* This complements the optional third parameter, `merge`, nicely: If you pass
```
Can we show a lazy resource loading example here? too

notwaldorf commented on app-localize-behavior.html:
```diff
@@ -52,6 +52,33 @@
+* This complements the optional third parameter, `merge`, nicely: If you pass
+* `merge = true`, the fetched resources will be merged into any existing
+* resources rather than clobbering them. Object.assign is used for the merge,
+* but you can also pass your own merge function to use that instead.
```
"You can also pass your own merge function" -- this is no longer true, right?

notwaldorf replied:
Ohhhh i see. No, this is bad news bears (and clojure doesn't like it when one thing can have 2 types). I would recommend not giving people this option, and only having merge be boolean

jab commented on bower.json:
```diff
@@ -32,7 +32,7 @@
     "paper-styles": "PolymerElements/paper-styles#1 - 2",
     "test-fixture": "PolymerElements/test-fixture#^3.0.0-rc.1",
-    "web-component-tester": "^4.0.0",
+    "web-component-tester": "^6.0.0",
```
(((
needed to appease bower, otherwise it complains with:
```
Unable to find a suitable version for test-fixture, please choose one by typing one of the numbers below:
    1) test-fixture#^1.0.0 which resolved to 1.1.2 and is required by web-component-tester#4.3.6
    2) test-fixture#^3.0.0-rc.1 which resolved to 3.0.0-rc.1 and is required by app-localize-behavior)
```
)))

notwaldorf replied:
🙌

jab commented:
I think I addressed all the issues we discussed. Squashed into a single commit that CI is currently passing for.

Thanks so much for reviewing @notwaldorf! Please let me know if you spot anything else I should do before this can be merged and I'll be happy to.

notwaldorf commented on app-localize-behavior.html:
```diff
@@ -247,9 +287,34 @@
+      }
+    }
+    if (this.setProperties) {
+      this.setProperties(propertyUpdates);
```
wait, what is `setProperties`? Is this JavaScript magic I don't know about?

jab replied:
new in polymer 2! https://www.polymer-project.org/2.0/docs/devguide/model-data#set-property

notwaldorf replied:
oh TIL it's Polymer magic: https://www.polymer-project.org/2.0/docs/devguide/model-data#set-property

jab replied:
jinx!

jab commented:
Thanks for the great review and for putting up with all my back-and-forth!
